### PR TITLE
Support for EXT_mesh_gpu_instancing extension

### PR DESCRIPTION
**Pull request by mvaligursky**
Fixes #https://github.com/playcanvas/engine/issues/3526

Example:
image

### CODE REVIEW

File: src/scene/mesh-instance.js [hunk 1]
```diff
@@ -58,12 +58,26 @@ class InstancingData {
     /** @type {VertexBuffer|null} */
     vertexBuffer = null;
 
+    /**
+     * True if the vertex buffer is destroyed when the mesh instance is destroyed.
+     *
+     * @type {boolean}
```
Comment (willeastcott) [inline src/scene/mesh-instance.js:64]: Should this be `@private`?

Comment (willeastcott) [reply src/scene/mesh-instance.js:64]: Ah, you reference it in `glb-container-resource.js`. So maybe `@ignore` instead?

**Review by willeastcott — APPROVED** [review]

File: src/scene/mesh-instance.js (hunk 1 above)
Comment (mvaligursky) [reply src/scene/mesh-instance.js:64]: we removed all `@ignore` tags from non-exported classes.

Comment (willeastcott) [reply src/scene/mesh-instance.js:64]: Oh, this is `InstancingData`. I thought it was `MeshInstance`. OK, forget it.

**Pull request merged**